src/QirRuntime/public/TracerTypes.hpp

```diff
@@ -10,8 +10,9 @@ namespace Quantum
     using OpId = int;
     using Time = int;
     using Duration = int;
-    using LayerId = size_t;
+    using LayerId = int64_t;
 
-    constexpr LayerId INVALID = std::numeric_limits<size_t>::max();
+    constexpr LayerId INVALID = std::numeric_limits<int64_t>::min();
+    constexpr LayerId REQUESTNEW = std::numeric_limits<int64_t>::max();

```

Comment:
Consider
    constexpr LayerId INVALID = std::numeric_limits<**LayerId**>::min();
    constexpr LayerId REQUESTNEW = std::numeric_limits<**LayerId**>::max();
    
If one replaces the type (`int64_t`) behind the `LayerId`, but forgets to replace the same type for `INVALID` and/or `REQUESTNEW`, then the behavior can become unexpected.